src/System.Threading.AccessControl/src/System/Security/AccessControl/EventWaitHandleSecurity.cs

```diff
@@ -0,0 +1,274 @@
+        {
+            get { return (EventWaitHandleRights)base.AccessMask; }
+        }
+    }
+
+
+    public sealed class EventWaitHandleSecurity : NativeObjectSecurity
+    {
+        public EventWaitHandleSecurity()
+            : base(true, ResourceType.KernelObject)
+        {
+        }
+
+        [System.Security.SecurityCritical]  // auto-generated
+        internal EventWaitHandleSecurity(String name, AccessControlSections includeSections)
+            : base(true, ResourceType.KernelObject, name, includeSections, _HandleErrorCode, null)
+        {
+            // Let the underlying ACL API's demand unmanaged code permission.
+        }
+
+        [System.Security.SecurityCritical]  // auto-generated
+        internal EventWaitHandleSecurity(SafeWaitHandle handle, AccessControlSections includeSections)
+            : base(true, ResourceType.KernelObject, handle, includeSections, _HandleErrorCode, null)
+        {
+            // Let the underlying ACL API's demand unmanaged code permission.
+        }
+
+        [System.Security.SecurityCritical]  // auto-generated
+        private static Exception _HandleErrorCode(int errorCode, string name, SafeHandle handle, object context)
+        {
+            System.Exception exception = null;
+
+            switch (errorCode)
+            {
+                case Interop.mincore.ERROR_INVALID_NAME:
+                case Interop.mincore.ERROR_INVALID_HANDLE:
+                case Interop.mincore.ERROR_FILE_NOT_FOUND:
+                    if ((name != null) && (name.Length != 0))
+                        exception = new WaitHandleCannotBeOpenedException(SR.Format(SR.WaitHandleCannotBeOpenedException_InvalidHandle, name));
+                    else
+                        exception = new WaitHandleCannotBeOpenedException();
+                    break;
+
+                default:
+                    break;
+            }
+
+            return exception;
+        }
+
+        public override AccessRule AccessRuleFactory(IdentityReference identityReference, int accessMask, bool isInherited, InheritanceFlags inheritanceFlags, PropagationFlags propagationFlags, AccessControlType type)
+        {
+            return new EventWaitHandleAccessRule(identityReference, accessMask, isInherited, inheritanceFlags, propagationFlags, type);
+        }
+
+        public override AuditRule AuditRuleFactory(IdentityReference identityReference, int accessMask, bool isInherited, InheritanceFlags inheritanceFlags, PropagationFlags propagationFlags, AuditFlags flags)
+        {
+            return new EventWaitHandleAuditRule(identityReference, accessMask, isInherited, inheritanceFlags, propagationFlags, flags);
+        }
+
+        internal AccessControlSections GetAccessControlSectionsFromChanges()
+        {
+            AccessControlSections persistRules = AccessControlSections.None;
+            if (AccessRulesModified)
+                persistRules = AccessControlSections.Access;
```

Comment:
It doesn't matter technically, but for consistency it'd be nice if this were ```|=``` instead of ```=```.